### PR TITLE
Fix issue#38

### LINKED ISSUE #38: undefined local variable or method `construct_relation_for_association_calculations' for #<ActiveRecord::Relation []>
**Reproduction steps:**
- Rails 5.2.2.1
- `includes` then `pluck_all`
  ```ruby
  User.where(id: 1).includes(:posts).pluck_all(:id)
  ```

It seems that `construct_relation_for_association_calculations`  was removed in [this commit](https://github.com/rails/rails/commit/bb5b672ac6b4dd694a3f706d98d82f3000ce18cd#diff-1cabaf53664d448a5cbfcf396cd842b1)

### PR DESCRIPTION
**Pull request by MasashiYokota**
Hi. I've fixed #38 based on [this RoR code](https://github.com/rails/rails/blob/350ad01b815979f492f5f359991c9be707d5e3b5/activerecord/lib/active_record/relation/calculations.rb#L187). 
Please review this PR.
Thank you.

### CODE REVIEW

File: lib/pluck_all/models/active_record_extension.rb [hunk 1]
```diff
@@ -43,7 +43,10 @@ def to_sql_column_name
   else
     def pluck_all(*column_names, cast_uploader_url: true)
       column_names.map!(&to_sql_column_name)
-      return construct_relation_for_association_calculations.pluck_all(*column_names) if has_include?(column_names.first)
+      if has_include?(column_names.first)
+        relation = apply_join_dependency
+        return relation.pluck(*column_names)
```
Comment (khiav223577) [inline lib/pluck_all/models/active_record_extension.rb]: It should use `pluck_all` method here to get hash results.
```suggestion
        return relation.pluck_all(*column_names)
```
(https://github.com/khiav223577/pluck_all/compare/fix/test_pluck_all_with_includes)

Comment (khiav223577) [reply lib/pluck_all/models/active_record_extension.rb]: I've added some test cases to test this line. 
You could merge this branch [fix/test_pluck_all_with_includes](https://github.com/khiav223577/pluck_all/compare/fix/test_pluck_all_with_includes) to make sure if your modification fixes the bug or not :)

Comment (MasashiYokota) [reply lib/pluck_all/models/active_record_extension.rb]: @khiav223577 
Sorry for my careless mistake.
And thank you for your comments and adding some test cases for this patch.

Thanks to the test cases, I've found and fixed some bugs caused by differences of Rails versions.
Thank you very much.

**Comment by khiav223577** [comment]
@MasashiYokota 
Thanks for your contribution 💯

**Comment by MasashiYokota** [comment]
@khiav223577 
Thank you very much 😄 

I have a favor to ask of you. This patch is necessary for my project. Would you release it ?

**Comment by khiav223577** [comment]
I released 2.0.4. You could see the [CHANGELOG](https://github.com/khiav223577/pluck_all/blob/master/CHANGELOG.md#v204-20190404).

**Comment by MasashiYokota** [comment]
Thank you so much 😄

**Pull request merged**